client/src/views/page-components/account/AccountHistoryModal/AccountHistoryModal.ts

```diff
@@ -0,0 +1,388 @@
+import { $, createDOMWithSelector, removeOthersClassList } from '@src/utils/helper';
+import './AccountHistoryModal.scss';
+import handleEvent from '@src/utils/handleEvent';
+import { categoryList, matchCategoryAndImg } from '@src/static/constants';
+import PayMethods from '@src/views/components/PayMethods/PayMethods';
+import { samplePay } from '@src/dummyData';
+
+const slideOutTime: number = 1300;
+const dateInputMaxLength: number = 8;
+const moneyInputMaxLength: number = 9;
+const moneyInputMinLength: number = 0;
+const alertShowTime: number = 2000;
+
+/**
+    추가적으로 고려할 부분 : 
+    1.수입 지출!
+    2.윤년 고려!
+ */
+export default class AccountHistoryModal {
+  state: any;
+  choicedCategoryIndex: number = 0;
+  payMethod: any;
+  modal: any;
+  modalWrapper;
+  dateInput;
+  moneyInput;
+  dateValueValidation: boolean = false;
+  moneyValueValidation: boolean = false;
+  choicedCategoryName: string = '';
+  type: string = '';
+
+  constructor() {
+    handleEvent.subscribe('createhistorymodal', (e: CustomEvent) => {
+      this.setState(e.detail); // undefined ( 애초에 가져올만한 정보는 결제수단목록뿐) !! 값 확인하기 !!
+
+      this.modalWrapper = createDOMWithSelector('div', '.account-history-wrapper');
+      this.render();
+
+      const payMethodForm = $('.history-form__pay-method');
+      this.dateInput = $('.history-form__date');
+      this.moneyInput = $('.history-form__money');
+
+      this.payMethod = new PayMethods({ parent: payMethodForm, state: samplePay }); // 결제수단의 정보 갖고있어야함!
+      this.modalWrapper.addEventListener('click', this.onClickHandler.bind(this));
+      this.modalWrapper.addEventListener('keyup', this.onKeyUpHandler.bind(this));
+      this.dateInput.addEventListener('focusout', this.onFocusOutDateInputHandler.bind(this));
+      this.moneyInput.addEventListener('focusout', this.onFocusOutMoneyInputHandler.bind(this));
```

Comment:
상위 요소에서 이벤트를 뿌려줘도 될것 같습니다~

```suggestion
      this.modalWrapper.addEventListener('focusout', ({target}) => handleFocusOutOnModalWrapper(target));
      
      const handleFocusOutOnModalWrapper = (target) => {
        if(target.closest('.date-input')) this.onFocusOutDateInputHandler.bind(this);
        if(target.closest('money-input')) this.onFocusOutMoneyInputHandler.bind(this);
      }
```